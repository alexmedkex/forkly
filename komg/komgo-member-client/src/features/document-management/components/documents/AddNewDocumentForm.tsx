import { Profile } from '../../../../store/common/types'
import { Document } from '../../../../features/document-management/store'
import { ApplicationState } from '../../../../store/reducers'
import { Formik, FormikProps } from 'formik'
import * as React from 'react'
import { connect } from 'react-redux'
import { Form } from 'semantic-ui-react'
import * as Yup from 'yup'
import * as path from 'path'

import { Category, CreateDocumentRequest, DocumentType } from '../../store/types'
import { categoryIdValidation, DocumentCategoryField } from './new-document-components/DocumentCategoryField'
import { DocumentNameField, nameValidation } from './new-document-components/DocumentNameField'
import { DocumentTypeField, typeIdValidation } from './new-document-components/DocumentTypeField'
import { FileUploadField, fileValidation } from './new-document-components/FileUploadField'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'

export interface Props {
  documents: Document[]
  categories?: Category[]
  documentTypes: DocumentType[]
  profile: Profile
  preselectedCategory: string
  preselectedDocumentType: string
  documentTypeDisabled?: boolean
  handleSubmit(formData: CreateDocumentRequest): void
  toggleSubmit?(enable: boolean): void
}

interface State {
  fileName: string
}

interface NewDocumentForm {
  name: string
  categoryId?: string
  typeId: string
  file: File | null
  fileName: string
}

export class AddNewDocumentForm extends React.Component<Props, State> {
  private validationSchema

  constructor(props: Props) {
    super(props)

    this.validationSchema = this.formValidationSchema()

    this.onFileNameUpdate = this.onFileNameUpdate.bind(this)
  }

  render() {
    return (
      <Formik
        initialValues={{
          name: '',
          categoryId: this.props.preselectedCategory || undefined,
          typeId:
            this.getPreselectedDocumentType(
              this.props.preselectedDocumentType,
              this.props.documentTypes,
              this.props.preselectedCategory
            ) || '',
          file: null,
          fileName: ''
        }}
        validationSchema={this.validationSchema}
        onSubmit={(values, actions) => {
          this.props.handleSubmit({
            name: values.name,
            categoryId: values.categoryId,
            documentTypeId: values.typeId,
            file: values.file!,
            creator: {
              firstName: this.props.profile.firstName,
              lastName: this.props.profile.lastName,
              companyId: this.props.profile.company
            }
          })
        }}
        render={(props: FormikProps<NewDocumentForm>) => {
          const { values, handleSubmit } = props

          const categoryId = this.props.categories ? values.categoryId : undefined
          this.props.toggleSubmit(props.isValid) // TODO: this logic should be refactored and moved from render

          return (
            <Form onSubmit={handleSubmit} id="submit-form" className="add-new-document">
              <DocumentNameField formik={props} />

              {this.props.categories && (
                <DocumentCategoryField
                  formik={props}
                  categories={sortCategories(this.props.categories)}
                  preselectedCategory={this.props.preselectedCategory}
                />
              )}

              <DocumentTypeField
                formik={props}
                categoryId={categoryId}
                documentTypes={sortDocumentTypes(this.props.documentTypes)}
                preselectedDocumentType={this.getPreselectedDocumentType(
                  this.props.preselectedDocumentType,
                  this.props.documentTypes,
                  values.categoryId
                )}
                disabled={this.props.documentTypeDisabled}
              />

              <FileUploadField formik={props} onFileNameUpdate={this.onFileNameUpdate} />
            </Form>
          )
        }}
      />
    )
  }

  /**
   * A document type is preselected if:
   * - It is passed explicitly as a prop or if it is.
   * - It is the only document type available for that category, in this case it is also preselected and disabled.
   */
  getPreselectedDocumentType(
    preselectedDocumentType: string,
    documentTypes: DocumentType[],
    categoryId: string
  ): string {
    const filteredDocumentTypes: DocumentType[] = documentTypes.filter(x => x.category.id === categoryId)
    if (preselectedDocumentType) {
      return preselectedDocumentType
    } else if (filteredDocumentTypes.length === 1) {
      return filteredDocumentTypes[0].id
    } else {
      return ''
    }
  }

  // To get document extension to build full name
  onFileNameUpdate(fileName: string) {
    this.setState({
      fileName
    })
  }

  private formValidationSchema(): any {
    const fields = {
      name: this.validateDocumentName(),
      ...typeIdValidation,
      ...fileValidation
    }

    return this.props.categories
      ? Yup.object().shape({
          ...fields,
          ...categoryIdValidation
        })
      : Yup.object().shape(fields)
  }

  private validateDocumentName() {
    return Yup.string()
      .required('Document title is required')
      .test(
        'unique-name',
        'A document with the same name already exists. Choose another name.',
        this.noDocumentWithTheSameName.bind(this)
      )
  }

  private noDocumentWithTheSameName(name: string): boolean {
    if (!this.state || !this.state.fileName) {
      return true
    }

    const docNameWithExt = `${name}${path.extname(this.state.fileName)}`
    const foundDocument = this.props.documents.find(doc => doc.name === docNameWithExt)
    return !foundDocument
  }
}

const mapStateToProps = (state: ApplicationState) => ({
  profile: state.get('uiState').get('profile')
})

export default connect(mapStateToProps, {})(AddNewDocumentForm)
