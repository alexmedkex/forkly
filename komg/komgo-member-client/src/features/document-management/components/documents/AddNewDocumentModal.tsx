import * as React from 'react'
import { Modal, Button } from 'semantic-ui-react'

interface Props {
  visible: boolean
  title: string
  children: any
  toggleVisible(): void
}

interface State {
  submitButtonDisabled: boolean
}

class AddNewDocumentModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = { submitButtonDisabled: true }
    this.toggleSubmitButton = this.toggleSubmitButton.bind(this)
  }

  toggleSubmitButton(option: boolean) {
    if (!option !== this.state.submitButtonDisabled) {
      this.setState({ submitButtonDisabled: !option })
    }
  }

  render() {
    return (
      <Modal open={this.props.visible} centered={true} style={{ top: 'unset' }}>
        <Modal.Header>{this.props.title}</Modal.Header>
        <Modal.Content>
          {React.Children.map(this.props.children, child => {
            return React.cloneElement(child, {
              // This code injects the toggleSubmit handler into the this.props.children
              toggleSubmit: this.toggleSubmitButton
            })
          })}
        </Modal.Content>
        <Modal.Actions>
          <Button className="ui button" role="button" onClick={this.props.toggleVisible}>
            Cancel
          </Button>
          <Button disabled={this.state.submitButtonDisabled} primary={true} content="Add document" form="submit-form" />
        </Modal.Actions>
      </Modal>
    )
  }
}

export default AddNewDocumentModal
