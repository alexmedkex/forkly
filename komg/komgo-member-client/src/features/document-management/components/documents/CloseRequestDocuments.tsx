import * as React from 'react'
import { Confirm } from 'semantic-ui-react'

interface Props {
  open: boolean
  content: string
  onConfirmClose(): void
  onToggleVisible(): void
}

class CloseRequestDocuments extends React.Component<Props> {
  handleConfirm = () => {
    this.props.onConfirmClose()
  }

  renderContent = (content: string) => {
    return { content }
  }

  render() {
    const { onToggleVisible, open, content } = this.props
    return (
      <Confirm
        open={open}
        cancelButton="Cancel"
        confirmButton="Confirm"
        header="Close request"
        content={this.renderContent(content)}
        onCancel={() => onToggleVisible()}
        onConfirm={() => this.handleConfirm()}
      />
    )
  }
}

export default CloseRequestDocuments
