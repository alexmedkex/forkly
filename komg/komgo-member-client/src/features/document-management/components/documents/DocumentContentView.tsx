import { ErrorMessage } from '../../../../components'
import { PERMITTED_MIME_TYPES } from '../../../document-management/utils/permittedMimeTypes'
import * as React from 'react'
import { Dimmer, Loader, Image } from 'semantic-ui-react'

import PdfContentView from './PdfContentView'

interface Props {
  documentContent: string
  documentType: string
  isLoadingContent: boolean
}

const DocumentContentView: React.SFC<Props> = (props: Props) => {
  if (props.isLoadingContent) {
    return (
      <Dimmer active={true}>
        <Loader indeterminate={true}>Loading</Loader>
      </Dimmer>
    )
  }

  if (!Object.values(PERMITTED_MIME_TYPES).includes(props.documentType)) {
    return <ErrorMessage title="Cannot view document" error="Document type not permitted" />
  }

  if (props.documentType === PERMITTED_MIME_TYPES.PDF_MIME_TYPE) {
    return <PdfContentView documentContent={props.documentContent} />
  }
  if (
    props.documentType === PERMITTED_MIME_TYPES.JPEG_MIME_TYPE ||
    props.documentType === PERMITTED_MIME_TYPES.PNG_MIME_TYPE
  ) {
    return <Image src={`data:${props.documentType};base64, ${props.documentContent}`} />
  }
  return <iframe height="500em" width="100%" src={`data:${props.documentType};base64, ${props.documentContent}`} />
}

export default DocumentContentView
