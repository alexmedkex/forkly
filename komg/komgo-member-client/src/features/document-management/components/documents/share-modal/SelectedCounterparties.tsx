import * as React from 'react'
import { Button, Modal, Popup, Flag } from 'semantic-ui-react'
import Highlighter from 'react-highlight-words'
import { Table, SPACES } from '@komgo/ui-components'
import styled from 'styled-components'

import { Counterparty } from '../../../../counterparties/store/types'
import { Document } from '../../../store/types'
import { buildCounterpartyWithDocumentsPickerItems } from './utils'
import {
  ActionsWrapper,
  TableAndSearchWrapper,
  CompanyTableItem
} from '../../../../credit-line/components/common/CounterpartyModalPicker'
import { WithSearchInput } from '../../../../../components'

export interface ICounterpartyWithDocuments {
  counterparty: Counterparty
  documents: Document[]
  isSelected: boolean
}

export interface ISelectedCounterpartiesProps {
  counterparties: ICounterpartyWithDocuments[]
  setSelectedCounterparties(staticIds: string[]): void
  onCancel(): void
  onConfirm(): void
}

interface IState {
  companies: ICompanyTableItemWithSharedDocs[]
}

export interface ICompanyTableItemWithSharedDocs extends CompanyTableItem {
  sharedDocuments: Document[]
}

export const CONTENT_CSS = { paddingTop: SPACES.SMALL, paddingBottom: SPACES.SMALL, height: '348px' }

class SelectedCounterparties extends React.Component<ISelectedCounterpartiesProps, IState> {
  static noCounterpartiesSelected = (props: ISelectedCounterpartiesProps) => {
    return props.counterparties.reduce<boolean>((prev, c) => prev && !c.isSelected, true)
  }

  constructor(props: ISelectedCounterpartiesProps) {
    super(props)

    this.state = {
      companies: buildCounterpartyWithDocumentsPickerItems(props.counterparties)
    }
  }

  handleSearchChanged = (search: string) => {
    const newCompanies = buildCounterpartyWithDocumentsPickerItems(this.props.counterparties).filter(member =>
      member.name.toLowerCase().includes(search.toLowerCase())
    )
    this.setState({ companies: newCompanies })
  }

  render() {
    const { companies } = this.state
    const { counterparties } = this.props

    const defaultSelected = counterparties.filter(c => c.isSelected).map(c => c.counterparty.staticId)

    return (
      <React.Fragment>
        <Modal.Header>
          <Modal.Description as="h2" content="Select a counterparty" style={{ marginBottom: 0 }} />
          <Modal.Description
            as="p"
            style={{ marginBlockStart: 0, fontSize: '1rem' }}
            content="Select the counterparties you wish to share documents with"
          />
        </Modal.Header>
        <Modal.Content style={CONTENT_CSS}>
          <TableAndSearchWrapper>
            <WithSearchInput
              onSearchChange={this.handleSearchChanged}
              render={search => (
                <Table
                  data={companies}
                  selectable={true}
                  dataTestId="name"
                  onCheckboxChange={this.props.setSelectedCounterparties}
                  defaultSelected={defaultSelected}
                  columns={[
                    {
                      accessor: 'name',
                      title: 'Counterparty',
                      width: 200,
                      cell: c => (
                        <Highlighter
                          highlightClassName="highlighted-text"
                          searchWords={[search]}
                          autoEscape={true}
                          textToHighlight={c.name}
                        />
                      )
                    },
                    {
                      accessor: 'country',
                      title: 'Country',
                      cell: c => (
                        <span>
                          <Flag name={c.country} /> {c.countryName}
                        </span>
                      )
                    },
                    { accessor: 'location', title: 'Location' },
                    {
                      accessor: 'sharedDocuments',
                      title: 'Documents shared',
                      align: 'right',
                      width: 142,
                      cell: c => (
                        <span>
                          {c.sharedDocuments.length ? (
                            <Popup
                              trigger={
                                <DocumentSharedPopupTrigger>{c.sharedDocuments.length}</DocumentSharedPopupTrigger>
                              }
                              position="right center"
                              content={
                                <div>
                                  <strong>Already shared</strong>
                                  <br />
                                  {c.sharedDocuments.map(d => d.name).join(', ')}
                                </div>
                              }
                            />
                          ) : (
                            '-'
                          )}
                        </span>
                      )
                    }
                  ]}
                />
              )}
            />
          </TableAndSearchWrapper>
        </Modal.Content>
        <Modal.Actions>
          <ActionsWrapper>
            <Button onClick={this.props.onCancel} data-test-id="cancel-button">
              Cancel
            </Button>
            <Button
              primary={true}
              disabled={SelectedCounterparties.noCounterpartiesSelected(this.props)}
              onClick={this.props.onConfirm}
              data-test-id="next-button"
            >
              Next
            </Button>
          </ActionsWrapper>
        </Modal.Actions>
      </React.Fragment>
    )
  }
}

const DocumentSharedPopupTrigger = styled.span`
  &:hover {
    cursor: pointer;
  }
`

export default SelectedCounterparties
