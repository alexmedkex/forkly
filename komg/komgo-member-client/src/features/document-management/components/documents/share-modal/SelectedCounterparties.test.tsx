import { shallow } from 'enzyme'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import mockCounterparties from '../../../../counterparties/store/mockData'
import { mockDocuments } from '../../../store/documents/mock-data'
import SelectedCounterparties, { ISelectedCounterpartiesProps } from './SelectedCounterparties'

describe('ConfirmShareStep component', () => {
  const mockProps: ISelectedCounterpartiesProps = {
    counterparties: mockCounterparties.counterparties.map(cp => ({
      counterparty: cp,
      documents: mockDocuments,
      isSelected: false
    })),
    setSelectedCounterparties: jest.fn(),
    onCancel: jest.fn(),
    onConfirm: jest.fn()
  }

  it('should match default snapshot', () => {
    expect(renderer.create(<SelectedCounterparties {...mockProps} />).toJSON()).toMatchSnapshot()
  })

  it('should filter company in state when handleSearchChanged is called', () => {
    const wrapper = shallow(<SelectedCounterparties {...mockProps} />)

    const instance = wrapper.instance() as SelectedCounterparties

    instance.handleSearchChanged('TestTest')

    expect(wrapper.state('companies')).toEqual([])
  })

  it('next button should be disabled if at least one counterpary is not selected', () => {
    const wrapper = shallow(<SelectedCounterparties {...mockProps} />)

    const nextButton = wrapper.find('[data-test-id="next-button"]')

    expect(nextButton.prop('disabled')).toBe(true)
  })

  it('next button should be enabled if at least one counterpary is selected', () => {
    const counterparties = [...mockProps.counterparties, { ...mockProps.counterparties[0], isSelected: true }]

    const wrapper = shallow(<SelectedCounterparties {...mockProps} counterparties={counterparties} />)

    const nextButton = wrapper.find('[data-test-id="next-button"]')

    expect(nextButton.prop('disabled')).toBe(false)
  })
})
