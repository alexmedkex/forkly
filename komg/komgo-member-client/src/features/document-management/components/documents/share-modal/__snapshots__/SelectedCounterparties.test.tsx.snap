// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmShareStep component should match default snapshot 1`] = `
Array [
  <div
    className="header"
  >
    <h2
      className="description"
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      Select a counterparty
    </h2>
    <p
      className="description"
      style={
        Object {
          "fontSize": "1rem",
          "marginBlockStart": 0,
        }
      }
    >
      Select the counterparties you wish to share documents with
    </p>
  </div>,
  .c3.c3.c3 {
  display: block;
  margin-left: 6px;
}

.c2.c2.c2 {
  display: block;
  margin: 0;
}

.c1.c1.c1 {
  width: 100%;
}

.c1.c1.c1 input {
  padding-right: 32px !important;
}

.c1.c1.c1 svg {
  top: 8px;
  right: 10px;
}

.c1.c1.c1 svg:hover {
  cursor: pointer;
}

.c0.c0.c0 .ReactTable {
  overflow-x: auto;
}

.c0.c0.c0 .ReactTable .rt-thead.-header {
  border-top: unset;
}

.c0.c0.c0 .ReactTable .rt-tbody {
  max-height: 240px;
  overflow-y: auto;
  overflow-x: hidden;
}

.c0.c0.c0 .ReactTable .rt-tbody::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(255,255,255,0.3);
  background-color: #c0cfde;
}

.c0.c0.c0 .ReactTable .rt-tbody::-webkit-scrollbar {
  width: 2px;
  background-color: #c0cfde;
}

.c0.c0.c0 .ReactTable .rt-tbody::-webkit-scrollbar-thumb {
  background-color: #5d768f;
  border: 2px solid #5d768f;
}

.c4:hover {
  cursor: pointer;
}

<div
    className="content"
    style={
      Object {
        "height": "348px",
        "paddingBottom": "16px",
        "paddingTop": "16px",
      }
    }
  >
    <div
      className="c0"
    >
      <div
        className="ui left icon input c1"
        data-test-id="search"
        open={false}
      >
        <i
          aria-hidden="true"
          className="search icon"
        />
        <input
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <div
        className="ReactTable"
        style={Object {}}
      >
        <div
          className="rt-table"
          role="grid"
        >
          <div
            className="rt-thead -header"
            style={
              Object {
                "minWidth": "602px",
              }
            }
          >
            <div
              className="rt-tr"
              role="row"
            >
              <div
                className="rt-th"
                data-test-id="id"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="columnheader"
                style={
                  Object {
                    "flex": "60 0 auto",
                    "maxWidth": "60px",
                    "width": "60px",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className=""
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "textAlign": "left",
                      }
                    }
                  >
                    <div
                      className="ui fitted checkbox c2"
                      onChange={[Function]}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <input
                        checked={false}
                        className="hidden"
                        onClick={[Function]}
                        readOnly={true}
                        tabIndex={0}
                        type="checkbox"
                      />
                      <label />
                    </div>
                  </span>
                </div>
              </div>
              <div
                className="rt-th rt-resizable-header -cursor-pointer"
                data-test-id="name"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="columnheader"
                style={
                  Object {
                    "flex": "200 0 auto",
                    "maxWidth": "200px",
                    "width": "200px",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="rt-resizable-header-content"
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "textAlign": "left",
                      }
                    }
                  >
                    Counterparty
                  </span>
                </div>
                <div
                  className="rt-resizer"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                />
              </div>
              <div
                className="rt-th rt-resizable-header -cursor-pointer"
                data-test-id="country"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="columnheader"
                style={
                  Object {
                    "flex": "100 0 auto",
                    "maxWidth": null,
                    "width": "100px",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="rt-resizable-header-content"
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "textAlign": "left",
                      }
                    }
                  >
                    Country
                  </span>
                </div>
                <div
                  className="rt-resizer"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                />
              </div>
              <div
                className="rt-th rt-resizable-header -cursor-pointer"
                data-test-id="location"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="columnheader"
                style={
                  Object {
                    "flex": "100 0 auto",
                    "maxWidth": null,
                    "width": "100px",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="rt-resizable-header-content"
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "textAlign": "left",
                      }
                    }
                  >
                    Location
                  </span>
                </div>
                <div
                  className="rt-resizer"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                />
              </div>
              <div
                className="rt-th rt-resizable-header -cursor-pointer"
                data-test-id="sharedDocuments"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="columnheader"
                style={
                  Object {
                    "flex": "142 0 auto",
                    "maxWidth": "142px",
                    "width": "142px",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="rt-resizable-header-content"
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "textAlign": "right",
                      }
                    }
                  >
                    Documents shared
                  </span>
                </div>
                <div
                  className="rt-resizer"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="rt-tbody"
            style={
              Object {
                "minWidth": "602px",
              }
            }
          >
            <div
              className="rt-tr-group"
              role="rowgroup"
            >
              <div
                className="rt-tr -odd"
                data-test-id="A Company"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="row"
                style={
                  Object {
                    "background": undefined,
                    "color": undefined,
                    "cursor": "inherit",
                  }
                }
              >
                <div
                  className="rt-td"
                  data-test-id="A Company-id"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "60 0 auto",
                      "maxWidth": "60px",
                      "overflow": "hidden",
                      "width": "60px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <div
                      className="ui fitted checkbox c3"
                      onChange={[Function]}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <input
                        checked={false}
                        className="hidden"
                        onClick={[Function]}
                        readOnly={true}
                        tabIndex={0}
                        type="checkbox"
                      />
                      <label />
                    </div>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="A Company-name"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "200 0 auto",
                      "maxWidth": "200px",
                      "overflow": "hidden",
                      "width": "200px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <span
                        className=""
                      >
                        A Company
                      </span>
                    </span>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="A Company-country"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "100 0 auto",
                      "maxWidth": null,
                      "overflow": "hidden",
                      "width": "100px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <i
                        className="country flag"
                      />
                       
                    </span>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="A Company-location"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "100 0 auto",
                      "maxWidth": null,
                      "overflow": "hidden",
                      "width": "100px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    city
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="A Company-sharedDocuments"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "142 0 auto",
                      "maxWidth": "142px",
                      "overflow": "hidden",
                      "width": "142px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "right",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <span
                        className="c4"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        3
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              className="rt-tr-group"
              role="rowgroup"
            >
              <div
                className="rt-tr -even"
                data-test-id="C Company"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="row"
                style={
                  Object {
                    "background": undefined,
                    "color": undefined,
                    "cursor": "inherit",
                  }
                }
              >
                <div
                  className="rt-td"
                  data-test-id="C Company-id"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "60 0 auto",
                      "maxWidth": "60px",
                      "overflow": "hidden",
                      "width": "60px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <div
                      className="ui fitted checkbox c3"
                      onChange={[Function]}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <input
                        checked={false}
                        className="hidden"
                        onClick={[Function]}
                        readOnly={true}
                        tabIndex={0}
                        type="checkbox"
                      />
                      <label />
                    </div>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="C Company-name"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "200 0 auto",
                      "maxWidth": "200px",
                      "overflow": "hidden",
                      "width": "200px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <span
                        className=""
                      >
                        C Company
                      </span>
                    </span>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="C Company-country"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "100 0 auto",
                      "maxWidth": null,
                      "overflow": "hidden",
                      "width": "100px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <i
                        className="country flag"
                      />
                       
                    </span>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="C Company-location"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "100 0 auto",
                      "maxWidth": null,
                      "overflow": "hidden",
                      "width": "100px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    city
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="C Company-sharedDocuments"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "142 0 auto",
                      "maxWidth": "142px",
                      "overflow": "hidden",
                      "width": "142px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "right",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <span
                        className="c4"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        3
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              className="rt-tr-group"
              role="rowgroup"
            >
              <div
                className="rt-tr -odd"
                data-test-id="N Company"
                onClick={[Function]}
                onContextMenu={[Function]}
                role="row"
                style={
                  Object {
                    "background": undefined,
                    "color": undefined,
                    "cursor": "inherit",
                  }
                }
              >
                <div
                  className="rt-td"
                  data-test-id="N Company-id"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "60 0 auto",
                      "maxWidth": "60px",
                      "overflow": "hidden",
                      "width": "60px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <div
                      className="ui fitted checkbox c3"
                      onChange={[Function]}
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <input
                        checked={false}
                        className="hidden"
                        onClick={[Function]}
                        readOnly={true}
                        tabIndex={0}
                        type="checkbox"
                      />
                      <label />
                    </div>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="N Company-name"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "200 0 auto",
                      "maxWidth": "200px",
                      "overflow": "hidden",
                      "width": "200px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <span
                        className=""
                      >
                        N Company
                      </span>
                    </span>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="N Company-country"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "100 0 auto",
                      "maxWidth": null,
                      "overflow": "hidden",
                      "width": "100px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <i
                        className="country flag"
                      />
                       
                    </span>
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="N Company-location"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "100 0 auto",
                      "maxWidth": null,
                      "overflow": "hidden",
                      "width": "100px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "left",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    city
                  </span>
                </div>
                <div
                  className="rt-td"
                  data-test-id="N Company-sharedDocuments"
                  onClick={[Function]}
                  role="gridcell"
                  style={
                    Object {
                      "flex": "142 0 auto",
                      "maxWidth": "142px",
                      "overflow": "hidden",
                      "width": "142px",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "block",
                        "overflow": "hidden",
                        "textAlign": "right",
                        "textOverflow": "ellipsis",
                      }
                    }
                  >
                    <span>
                      <span
                        className="c4"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        3
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="-loading"
        >
          <div
            className="-loading-inner"
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
    className="actions"
  >
    <div
      className="c0"
    >
      <button
        className="ui button"
        data-test-id="cancel-button"
        onClick={[Function]}
        role="button"
      >
        Cancel
      </button>
      <button
        className="ui primary disabled button"
        data-test-id="next-button"
        disabled={true}
        onClick={[Function]}
        role="button"
        tabIndex={-1}
      >
        Next
      </button>
    </div>
  </div>,
]
`;
